### ISSUE TITLE
Environment variables for sources are not respected POETRY_REPOSITORIES_XXX_URL

### ISSUE THREAD

**Issue by arichtman** [python-poetry/poetry#5992]
Hi all - I love Poetry and thanks for all the community has put into it so far.

```Bash
$ uname -a
Linux ed184cf034ba 5.10.102.1-microsoft-standard-WSL2 #1 SMP Wed Mar 2 00:30:59 UTC 2022 x86_64 GNU/Linux
$ cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
$ python3 --version
Python 3.9.2
$ poetry --version
Poetry version 1.1.14
$ env | grep -i poetry
POETRY_SOURCE_TEST_URL=https://test.pypi.org/simple
POETRY_TOOL_POETRY_SOURCE_TEST_URL=https://test.pypi.org/simple
POETRY_REPOSITORIES_TEST_URL=https://test.pypi.org/simple
POETRY_REPOSITORIES_TEST=https://test.pypi.org/simple
$ poetry install -vvv
Skipping virtualenv creation, as specified in config file.
Updating dependencies
Resolving dependencies...
   1: fact: env-repo-fail is 0.1.0
   1: derived: env-repo-fail
   1: fact: env-repo-fail depends on bprint (^0.1)
   1: selecting env-repo-fail (0.1.0)
   1: derived: bprint (>=0.1,<0.2)
   1: fact: no versions of bprint match >=0.1,<0.2
   1: conflict: no versions of bprint match >=0.1,<0.2
   1: !  bprint (^0.1) is satisfied by  bprint (^0.1)
   1: ! which is caused by "env-repo-fail depends on bprint (^0.1)"
   1: ! thus: version solving failed
   1: Version solving took 0.005 seconds.
   1: Tried 1 solutions.

  Stack trace:

  8  /usr/local/lib/python3.9/dist-packages/clikit/console_application.py:131 in run
      129│             parsed_args = resolved_command.args
      130│
    → 131│             status_code = command.handle(parsed_args, io)
      132│         except KeyboardInterrupt:
      133│             status_code = 1

  7  /usr/local/lib/python3.9/dist-packages/clikit/api/command/command.py:120 in handle
      118│     def handle(self, args, io):  # type: (Args, IO) -> int
      119│         try:
    → 120│             status_code = self._do_handle(args, io)
      121│         except KeyboardInterrupt:
      122│             if io.is_debug():

  6  /usr/local/lib/python3.9/dist-packages/clikit/api/command/command.py:171 in _do_handle
      169│         handler_method = self._config.handler_method
      170│
    → 171│         return getattr(handler, handler_method)(args, io, self)
      172│
      173│     def __repr__(self):  # type: () -> str

  5  /usr/local/lib/python3.9/dist-packages/cleo/commands/command.py:92 in wrap_handle
       90│         self._command = command
       91│
    →  92│         return self.handle()
       93│
       94│     def handle(self):  # type: () -> Optional[int]

  4  /usr/local/lib/python3.9/dist-packages/poetry/console/commands/install.py:71 in handle
       69│         self._installer.verbose(self._io.is_verbose())
       70│
    →  71│         return_code = self._installer.run()
       72│
       73│         if return_code != 0:

  3  /usr/local/lib/python3.9/dist-packages/poetry/installation/installer.py:103 in run
      101│         local_repo = Repository()
      102│
    → 103│         return self._do_install(local_repo)
      104│
      105│     def dry_run(self, dry_run=True):  # type: (bool) -> Installer

  2  /usr/local/lib/python3.9/dist-packages/poetry/installation/installer.py:235 in _do_install
      233│             )
      234│
    → 235│             ops = solver.solve(use_latest=self._whitelist)
      236│         else:
      237│             self._io.write_line("Installing dependencies from lock file")

  1  /usr/local/lib/python3.9/dist-packages/poetry/puzzle/solver.py:65 in solve
       63│         with self._provider.progress():
       64│             start = time.time()
    →  65│             packages, depths = self._solve(use_latest=use_latest)
       66│             end = time.time()
       67│

  SolverProblemError

  Because env-repo-fail depends on bprint (^0.1) which doesn't match any versions, version solving failed.

  at /usr/local/lib/python3.9/dist-packages/poetry/puzzle/solver.py:241 in _solve
      237│             packages = result.packages
      238│         except OverrideNeeded as e:
      239│             return self.solve_in_compatibility_mode(e.overrides, use_latest=use_latest)
      240│         except SolveFailure as e:
    → 241│             raise SolverProblemError(e)
      242│
      243│         results = dict(
      244│             depth_first_search(
      245│                 PackageNode(self._package, packages), aggregate_package_nodes
```

- **OS version and name**: Debian 11
- **Poetry version**: 1.1.14
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: [Gist](https://gist.github.com/arichtman/4e096f1cbae874ad5c82de5f39ebec36)

## Issue

Hi all - I love Poetry and thanks for all the community has put into it so far.

I am attempting to use Poetry in a CI environment, specifically, I wish to set the source package repository to an internal one with authentication. I have replicated the issue using `test.pypi.org` so that it may be more easily reproduced.
According to my reading of [documentation](https://python-poetry.org/docs/configuration/#using-environment-variables) and the [code](https://github.com/python-poetry/poetry/blob/bf885acdc9d01f8b842502ba832e31d438e51802/src/poetry/config/config.py#L191), environment variables set using syntax `"POETRY_REPOSITORIES_([A-Z_]+)_URL"` should be populated and used in configuration. However when that variable is set, I'm unable to `poetry add` or `poetry install`

**Comment by dimbleby** [comment]
the repositories are found [here](https://github.com/python-poetry/poetry/blob/c2c69a3cb2b86f23406a044878e58c4000f97e9c/src/poetry/factory.py#L79) but then discarded.

Should be straightforward to fix if anyone wants to submit an MR

**Comment by dimbleby** [comment]
duplicates #4880, duplicated by #6004

**Comment by abn** [comment]
@arichtman the code you are referring to was only introduced in https://github.com/python-poetry/poetry/commit/58db4247e6a1c85a249e813404acefc5a97eaffb as a fix for this issue. This is only available in `>1.2.0b2`.

Also note https://github.com/python-poetry/poetry/issues/6004#issuecomment-1194822105.
